Sources/PointFree/RequestContext.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import HttpPipeline
+import Prelude
+
+/// A value that wraps any given data with additional context that is useful for completing the
+/// request-to-response lifecycle.
+struct RequestContext<A> {
```

Comment:
This is nice! Seems like `RequestContext` is also `LayoutContext`: things that the page depends on no matter what. With different layouts it's a bit more amorphous. In the end a band-aid to approach row polymorphism? Wonder if revisiting some of that protocol union stuff you explored with GraphQL would pay off?

Reply:
ahhhh layout context! yeah with the lil I'm played around with layouts I was making wrapper types and the `contramap`ing on em to glue em together. we could rename later if that starts to make more sense